crates/protogen/proto/metastore/catalog.proto

```diff
@@ -49,8 +49,13 @@ message CatalogState {
 
   // Metadata for the deployment.
   DeploymentMetadata deployment = 3;
-
   // next: 4
+
+  // The version of the actual catalog implementation.
+  // This is incremented when there are incompatible changes to the physical
+  // catalog layout.
+  // Purposely set to a high number to avoid accidental collisions
+  optional uint32 catalog_version = 2047;
```

Comment:
might make sense to put this, just for readability at the top of the message definition (I think field number and ordering in the IDL doesn't need to be the same, but I might be wrong).